File: ETour-DotNet/Models/Schedule.cs
namespace ETour.Api.Models;

public class Schedule
{
    public int ScheduleId { get; set; }
    public int? TourId { get; set; }
    public Tour Tour { get; set; }
    public int? AvailableSeats { get; set; }
    public int? TotalSeats { get; set; }
    public string Status { get; set; }
    public DateOnly? StartDate { get; set; }
}

File: E-Tour/src/main/java/com/etour/entities/Schedule.java
package com.etour.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "schedule")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Schedule {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "schedule_id")
    @EqualsAndHashCode.Include
    private Integer scheduleId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tour_id")
    @ToString.Exclude
    private Tour tour;

    @Column(name = "available_seats")
    private Integer availableSeats;

    @Column(name = "total_seats")
    private Integer totalSeats;

    @Column(name = "status", length = 20)
    private String status;

    @Column(name = "start_date")
    private LocalDate startDate;
}
